Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-roleserversideviewservice', function (Y) {
+    "use strict";
+    /**
+     * Provides the role server side view service class
+     *
+     * @method ez-roleserversideviewservice
+     */
+    Y.namespace('eZ');
+
+    /**
+     * The Role Server Side View Service class.
+     *
+     * @namespace eZ
+     * @class RoleServerSideViewService
+     * @constructor
+     * @extends eZ.ServerSideViewService
+     */
+    Y.eZ.RoleServerSideViewService = Y.Base.create('roleServerSideViewService', Y.eZ.ServerSideViewService, [], {
+        initializer: function () {
+            this.on('*:contentDiscover', function (e) {
+                e.config.contentDiscoveredHandler = Y.bind(this._assignRole, this);
+            });
+        },
+
+        /**
+         * Assign the role to the Content items after the user has chosen them in
+         * the universal discovery widget
+         *
+         * @method _assignRole
+         * @protected
+         * @param {EventFacade} e
+         */
+        _assignRole: function (e) {
+            var data = e.target.get('data'),
+                userService = this.get('capi').getUserService(),
+                discoveryService = this.get('capi').getDiscoveryService(),
+                service = this;
+
+            this._assignRoleNotificationStarted(
+                data.roleId, data.roleName, e.selection
+            );
+
+            userService.loadRole(data.roleId, function (error, response) {
+                var role,
+                    tasks = new Y.Parallel(),
+                    end = tasks.add(function (err, response) {
+                        if (err) {
+                            service._error('tasks.add() failed: ' + err.message);
+                            return;
+                        }
+
+                        console.log('end'); // TODO just to let us know that it was picked, error handling should be here
```

Comment:
Can be removed I think (that was only temporary way to sign that we hit that function).
As there can be many objects selected for assign the role, maybe we could show user bit more information about that how many assigns was successful and also notify user about every failed try. Something like here: https://github.com/ezsystems/PlatformUIBundle/blob/master/Resources/public/js/views/services/plugins/ez-locationcreateplugin.js#L83
Depends on specification as I don't know if it was precisely described.